util/chplenv/chpl_gpu.py

```diff
@@ -237,8 +280,24 @@ def _validate_cuda_llvm_version_impl(gpu: gpu_type):
         )
 
 def _validate_rocm_llvm_version_impl(gpu: gpu_type):
-    if chpl_llvm.get() == 'bundled':
+    major_version = get_sdk_version().split('.')[0]
+
+    if major_version in ('4', '5') and chpl_llvm.get() == 'bundled':
         error("Cannot target AMD GPUs with CHPL_LLVM=bundled")
+    elif major_version == '6' and chpl_llvm.get() != 'bundled':
+        # once https://github.com/llvm/llvm-project/issues/65188 is resolved,
+        # ths branch check can removed
+        # assert(int(chpl_llvm.get_llvm_version()) >= 18)
+        error("Cannot target AMD GPUs with ROCm 6.x without CHPL_LLVM=bundled")
```

Comment:
As mentioned in Slack, I'd like for us to allow for `CHPL_LLVM=system` where we know that Spack has patched LLVM appropriately.

Can we please either convert this hard error to warning, or alternatively make it conditional on a "secret handshake" type envvar that says "yes I really know what I'm doing, this LLVM has been patched", eg `CHPL_LLVM_65188_PATCH=yes`

Reply:
I think I prefer the handshake approach. Warnings are easy to miss (especially if you just run `make` and walk away), and any LLVM 18 that is missing the patch will segfault on any AMD GPU compile. 

I've added an opt-out env as you suggest as `CHPL_LLVM_65188_PATCH`